android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt

```diff
@@ -254,6 +272,33 @@ open class QuestionnaireActivity : BaseMultiLanguageActivity(), View.OnClickList
     }
   }
 
+  private suspend fun setInitialExpression(questionnaire: Questionnaire) {
+    // TODO handle hierarchy and scope
+    // Immunization?code=1234
+    // %immunization.reference/code
+    questionnaire.item.flattened().forEach { item ->
+      item.initialExpression?.takeIf { it.language == "application/x-fhir-query" }
+        ?.let {expression->
+          // val resolvedExpression: String = ExpressionEvaluator::class
+            //.callFunction("createXFhirQueryFromExpression", expression, launchContexts) as String
+          val answerOptions: List<QuestionnaireItemInitialComponent> = QuestionnaireFragment::class
+            .callSuspendFunctionOnField(fragment, "viewModel", "loadAnswerExpressionOptions", item, expression) as List<QuestionnaireItemInitialComponent>
+          answerOptions
+        }
+        ?.let {
+          it
+            .let {
+              item.initial =
+                it.map {
+                  Questionnaire.QuestionnaireItemInitialComponent().apply {
+                    value = it.castToType(it)
+                  }
+                }
+            }
+        }
+    }
+  }
+
```

Comment:
@maimoonak can you comment what was your issue again? and what are you trying to achieve?
I can see that you're using reflection.

Reply:
The functionality of loading data of relevant type and use of choiceColumn and/or candidate expression and answer expression has already been implemented plus the use of launch context for deriving the data loading from database. Hence, we are using the existing method to do all the work of loading answers for initial expression, until we get this available into SDK.

https://build.fhir.org/ig/HL7/sdc/expressions.html#initialExpression "Most typically, it will draw on [launchContext](https://build.fhir.org/ig/HL7/sdc/expressions.html#launchContext) extension and [itemPopulationContext](https://build.fhir.org/ig/HL7/sdc/expressions.html#itemPopulationContext) extension to populate questionnaire elements based on information available from outside the QuestionnaireResponse.

Examples of use for Form Behavior might include defaulting one date or provider in a form to one entered earlier in the form, defaulting a selected location based on patient address, etc. For population, this is the primary mechanism of populating fields. Information passed in launchContext or queried from the health record is used to determine the initial value for the field. The response author then can review and potentially adjust the values"

https://build.fhir.org/ig/HL7/sdc/populate.html#expression-based-population "For [Full Population](https://build.fhir.org/ig/HL7/sdc/populate.html#mode), use the [questionnaire-initialExpression](https://build.fhir.org/ig/HL7/sdc/StructureDefinition-sdc-questionnaire-initialExpression.html) to cause the initial answer for the question to be set to the specified expression. This should always be a FHIRPath or CQL expression that resolves to an item of the appropriate type unless the element has a type of Reference, in which case the expression can ALSO be a FHIR query - and the item will be populated with a reference to the resource. Another exception is for an element of type .... 
For [Choice Selection](https://build.fhir.org/ig/HL7/sdc/populate.html#mode), use the [sdc-questionnaire-candidateExpression](https://build.fhir.org/ig/HL7/sdc/StructureDefinition-sdc-questionnaire-candidateExpression.html) to make available a list of possible answers for the user to choose from. As with initialExpression, this should always be a FHIRPath or CQL expression that resolves to an item of the appropriate type unless the element has a type of Reference, in which case the expression can be a FHIR query - and any selected item will be populated with a reference to that resource."

Reply:
so you're trying to populate initial selected choice answers using initialExpression, and FHIR Query.
the SDK doesn't support that.
so you need a method that's not public, hence you use reflection.

makes sense.

but I don't understand why you're not trying to implement this in the SDK, unless this is urgent?

Reply:
conclusion after the call:
it's takes a long time to merge in the SDK, as we may know